docroot/sites/its.uiowa.edu/modules/its_core/its_core.module

```diff
@@ -65,11 +70,67 @@ function its_core_entity_presave(EntityInterface $entity) {
       ]));
       $entity->setTitle($output);
 
+      // If we're saving an outage or service degradation,
+      // we also want to check and set our views cache invalidation date.
+      if ($entity->isPublished()) {
+        $cache_affected_targets = [];
+        foreach (['service-degradation', 'outage', 'planned-maintenance', 'service-announcement'] as $target) {
+          $cache_affected_targets[] = its_core_alert_type_tag_map($target);
+        }
+        if (in_array($entity->field_alert_category->target_id, $cache_affected_targets)) {
+          if ($entity instanceof ClosableInterface) {
+            // If the alert is already closed, we're not concerned
+            // with invalidating caches.
+            if (!$entity->isClosed()) {
+              $state_key = 'its_core_alerts_next_clear';
+              $next_clear = \Drupal::state()->get($state_key);
+              // If we don't have a current next clear time,
+              // or we need a sooner time, set it now.
+              if (is_null($next_clear) || $entity->field_alert_date?->end_value < $next_clear) {
+                \Drupal::state()->set($state_key, $entity->field_alert_date?->end_value);
+              }
+              // And invalidate the current cache to make sure
+              // any updates are reflected.
+              its_core_clear_custom_cache_tag('its-core-alerts');
+            }
+          }
+        }
+      }
+
       break;
 
   }
 }
 
+/**
+ * Implements hook_preprocess_fieldset().
+ */
+function its_core_preprocess_fieldset(&$variables) {
+  if (
```

Comment:
Wondering if we can avoid the regex, etc. and handle this similar to https://github.com/uiowa/uiowa/blob/main/docroot/sites/facilities.uiowa.edu/modules/facilities_core/facilities_core.module#L436

```
  if (isset($variables['element']['#type']) &&
    $variables['element']['#type'] === 'radios') {
    $admin_context = \Drupal::service('router.admin_context');
    if ($admin_context->isAdminRoute()) {
      $current_route = \Drupal::routeMatch();
      $node_type_data = $current_route?->getParameter('node_type');
      $node_type = $node_type_data?->get('type');

      if (isset($node_type) && $node_type === 'alert') {
        $variables['description_display'] = 'before';
      }
    }
  }
```

Reply:
<img src="https://qph.cf2.quoracdn.net/main-qimg-0e442e0f219ba1198e551db2f8a3ea97.webp"/>

Reply:
Shared a DM with Alan about this. Initially there was some concern of the reduced specificity of not singling out this one field, "Category". I think that we have scoped to the site, content type, field type and that should be enough. If another radios field is added to the form, it will be consistent between the two. Scoping on a field label is a bit shaky as a label change would stop this from functioning.